File: emulinker/src/main/java/org/emulinker/kaillera/model/event/UserDroppedGameEvent.java
package org.emulinker.kaillera.model.event;

import org.emulinker.kaillera.model.*;

public class UserDroppedGameEvent implements GameEvent {
  private KailleraGame game;
  private KailleraUser user;
  private int playerNumber;

  public UserDroppedGameEvent(KailleraGame game, KailleraUser user, int playerNumber) {
    this.game = game;
    this.user = user;
    this.playerNumber = playerNumber;
  }

  @Override
  public String toString() {
    return "UserDroppedGameEvent";
  }

  @Override
  public KailleraGame getGame() {
    return game;
  }

  public KailleraUser getUser() {
    return user;
  }

  public int getPlayerNumber() {
    return playerNumber;
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/event/UserDroppedGameEvent.kt
package org.emulinker.kaillera.model.event

import org.emulinker.kaillera.model.KailleraGame
import org.emulinker.kaillera.model.KailleraUser

data class UserDroppedGameEvent(
    override val game: KailleraGame, val user: KailleraUser, val playerNumber: Int
) : GameEvent {
  override fun toString(): String {
    return "UserDroppedGameEvent"
  }
}
